src/transform/src/redundant_join.rs

```diff
@@ -387,20 +405,21 @@ impl RedundantJoin {
                     // been a problem in `Union`, where we might report
                     // that the union of positive and negative records is
                     // "exact": cancellations would make this false.
-                    let mut result = self.action(input, lets)?;
+                    let mut result = self.action(input, ctx)?;
                     for prov in result.iter_mut() {
                         prov.exact = false;
                     }
                     Ok(result)
                 }
 
-                MirRelationExpr::ArrangeBy { input, .. } => self.action(input, lets),
+                MirRelationExpr::ArrangeBy { input, .. } => self.action(input, ctx),
             }
         })?;
         // println!("{}", relation.pretty());
         // println!("result = {result:?}");
         // println!("lets: {lets:?}");
         // println!("---------------------");
+        result.retain(|info| !info.is_trivial());
```

Comment:
Nit: Move before comment? Or maybe the comment should be removed?

Reply:
I'm keeping the comment because it's useful when debugging and understanding this (which I had to twice in the last 3 months).